text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,195 @@
+6. (If compiled) Upload the dependency to the dependency GCP bucket
+7. (If compiled) Add the GCP bucket access URI and the SHA256 of the dependency to the
+    metadta `URI` and `SHA256` fields
+8. Update the `buildpack.toml` with the new versions and metadata
+
+This new plan eliminates extra steps of storing metadata in a file for later
+use, and then separately updating the dependencies on a timer.
+
+### Workflows and Actions
+Pending the Phase 1 RFC is accepted, and it's decided dependency logic will
+live alongside the buildpack, workflows will be rolled out to all of the
+buildpacks from
+[paketo-buildpacks/github-config](https://github.com/paketo-buildpacks/github-config)
+inside the `implementation/.github/workflows` location.
+
+Actions will live in the github-config repository as well, under a
+directory named `dependency`, inside of the `actions` directory.
+
+The github-config repository `CODEOWNERS` file will be updated to set the
+`@paketo-buildpacks/dependencies-maintainers` to the owners of the
+`actions/dependency` and `implementation/.github/workflows/<all
+dependency-related workflows>`.
+
+
+Each of the step s outlined in the overview will be converted into workflow
+steps.
+
+Workflow 1:
+1. Picking up new versions with buildpack code will be a workflow step, which
+   runs on a timer at least once per day.
+
+2. Figuring out which versions will be added to the `buildpack.toml` will be
+   rolled into an action that outputs versions. It will leverage parts of the
+   existing `jam update-dependencies` logic.
+
+3. For individual versions, a Github Actions [`matrix
+   strategy`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy)
+   will be used to iterate over each version to be added, to generate metadata
+   using the code from within the buildpacks.
+
+   The metadata generation code from the buildpack will contain logic to spit
+   out "variants" for each dependency version, depending on the stacks
+   (OS/architectures) supported by the buildpack.
+
+   For example, if the dependency is used from source but has two different
+   artifacts for ARM64 and AMD64 variants, and the buildpack supports both
+   architectures, then the metadata generation code should spit out two batches of
+   metadata for each new version. One with the AMD64 upstream source URI/SHA256, and
+   the other with the ARM64 upstream URI/source SHA256.
+
+5. In the same matrix loop, dependency compilation will be kicked off with the
+   code from the buildpacks on the basis of whether the metadata contains the
+   `SHA256` and `URI` of the dependency.
+
+6. A smoke test will be run against the dependency in both the
+   compiled/non-compiled cases via a step that simply runs the test from
+   `<buildpack>/dependency/test`.
+
+7. (If compiled) The dependency will be uploaded via an action
+
+8. (If compiled) The metadata is updated with the SHA256 and GCP bucket URI in
+   a workflow step
```

Comment:
Have we considered making this a separate workflow that can be used in the rare cases where it is needed? Then the update/metadata workflow could be the same in all cases, the only difference being it looks for our compiled dependency instead of an upstream artifact.

Reply:
I'm not sure I understand why this should be a separate workflow step. I think it can pretty easily just be coupled with the compilation job in the workflow, such that if compilation/artifact upload step runs, then the SHA256 and URI are just updated in the metadata. It seems pretty coupled, I can't wrap my head around why we'd need a separate mechanism to do this.

The other steps such as original metadata generation, and the `buildpack.toml` update will be the same in every case